inference-engine/samples/speech_sample/main.cpp

```diff
@@ -982,19 +865,28 @@ int main(int argc, char* argv[]) {
                 int i = 0;
                 for (auto& ptrInputBlob : ptrInputBlobs) {
                     if (ptrInputBlob->size() != numFrameElementsInput[i++] * batchSize) {
-                        throw std::logic_error("network input size(" + std::to_string(ptrInputBlob->size()) + ") mismatch to ark file size (" +
+                        throw std::logic_error("network input size(" + std::to_string(ptrInputBlob->size()) + ") mismatch to input file size (" +
                                                std::to_string(numFrameElementsInput[i - 1] * batchSize) + ")");
                     }
                 }
 
                 ptrScores.resize(numFrames * numScoresPerFrame * sizeof(float));
                 if (!FLAGS_r.empty()) {
-                    /** Read ark file with reference scores **/
+                    /** Read file with reference scores **/
+                    BaseFile* fileReferenceScores;
+                    auto exReferenceScoresFile = fileExt(FLAGS_r);
+                    if (exReferenceScoresFile == "ark") {
+                        fileReferenceScores = new ArkFile();
```

Comment:
We don't need to allocate the instance in the memory heap.

Reply:
Now allocate on stack.